install-deps.sh

```diff
@@ -96,8 +96,6 @@ ENDOFKEY
 	$SUDO env DEBIAN_FRONTEND=noninteractive apt-get update -y || true
 	$SUDO env DEBIAN_FRONTEND=noninteractive apt-get install -y g++-${new}
     fi
-
-    in_jenkins && echo "CI_DEBUG: End ensure_decent_gcc_on_ubuntu() in install-deps.sh"
```

Comment:
why is `in_jenkins` considered as an `early return`. it is the *last* statement in this function, and it does not return, IMO.

Reply:
This is one of `set -e` sharp edges.  In most cases, `&&` would suppress potential non-zero status from (as it does in a similar statement at the top of `ensure_decent_gcc_on_ubuntu`, for example).  However, in the absence of explicit `return`, the status of the function is always the status of the last executed command and `&&` doesn't change that.  If not on a Jenkins node, the last executed command here would be `in_jenkins`, causing `ensure_decent_gcc_on_ubuntu` to return with non-zero status.  `set -e` then happily axes the rest of the script.